core/src/block.rs

```diff
@@ -243,21 +226,52 @@ mod valid {
 
 /// Signed block related structures and implementations.
 mod signed {
+    use iroha_version::{declare_versioned_with_scale, version_with_scale};
+
     use super::*;
 
-    /// After receiving first signature, [`ValidBlock`] can transform into [`SignedBlock`].
-    #[derive(Debug, Clone)]
+    declare_versioned_with_scale!(VersionedSignedBlock 1..2, Debug, Clone, iroha_macro::FromVariant);
+
+    impl VersionedSignedBlock {
+        #[inline]
+        #[allow(missing_docs)]
+        pub const fn as_v1(&self) -> &SignedBlock {
+            match self {
+                Self::V1(v1) => v1,
+            }
+        }
+
+        #[inline]
+        #[allow(missing_docs)]
+        pub fn as_mut_v1(&mut self) -> &mut SignedBlock {
+            match self {
+                Self::V1(v1) => v1,
+            }
+        }
+
+        #[inline]
+        #[allow(missing_docs)]
+        pub fn into_v1(self) -> SignedBlock {
+            match self {
+                Self::V1(v1) => v1,
+            }
+        }
+    }
+
+    /// Signed block
+    #[version_with_scale(n = 1, versioned = "VersionedSignedBlock")]
+    #[derive(Debug, Clone, Decode, Encode)]
     pub struct SignedBlock {
```

Comment:
Shouldn't it be renamed as `Signed` like all other `Block`-related stuff in this module?

Reply:
it goes like this: `BlockBuilder<Pending>` - >`BlockBuilder<Chained>` -> `SignedBlock`